src/main/java/org/broadinstitute/hellbender/utils/read/GoogleGenomicsReadToGATKReadAdapter.java

```diff
@@ -703,4 +733,105 @@ private void workAroundHttpClientCloneBug() {
+        sb.append(Integer.toString(ReadUtils.getSAMFlagsForRead(this)));
+        sb.append(SAM_FIELD_SEPARATOR);
+        sb.append(null == contigName ? SAMRecord.NO_ALIGNMENT_REFERENCE_NAME : contigName);
+        sb.append(SAM_FIELD_SEPARATOR);
+        sb.append(Integer.toString(getStart()));
+        sb.append(SAM_FIELD_SEPARATOR);
+        sb.append(Integer.toString(getMappingQuality()));
+        sb.append(SAM_FIELD_SEPARATOR);
+        sb.append(getCigar().toString());
+        sb.append(SAM_FIELD_SEPARATOR);
+
+        if (!isPaired()) {
+            sb.append("=");
+        }
+        else {
+            final String mateContigName = getMateContig();
+            if (null == mateContigName) {
+                sb.append(SAMRecord.NO_ALIGNMENT_REFERENCE_NAME);
+            }
+            else if (null != contigName &&
+                    !contigName.equals(SAMRecord.NO_ALIGNMENT_REFERENCE_NAME) &&
+                    contigName.equals(mateContigName)) {
+                sb.append("=");
+            }
+            else {
+                sb.append(mateContigName);
+            }
+        }
+        sb.append(SAM_FIELD_SEPARATOR);
+        sb.append(Integer.toString(getMateStart()));
+        sb.append(SAM_FIELD_SEPARATOR);
+        sb.append(Integer.toString(getFragmentLength())); // getInferredInsertSize
+        sb.append(SAM_FIELD_SEPARATOR);
+
+        final byte bases[] = getBases();
+        if (null == bases || 0 == bases.length) {
+            sb.append(ReadConstants.NULL_SEQUENCE_STRING);
+        }
+        else {
+            sb.append(StringUtil.bytesToString(bases));
+        }
+
+        sb.append(SAM_FIELD_SEPARATOR);
+        sb.append(getBaseQualityString());
+
+        if (null != genomicsRead.getInfo()) {
+            TreeSet<String> sortedAttributes = new TreeSet<String>();
+            Set<String> keys = genomicsRead.getInfo().keySet();
+            sortedAttributes.addAll(keys);
+            // Query for any additional attributes that are not stored in set returned by read.getInfo
+            sortedAttributes.addAll(querySAMAttributes());
+            for (String key: sortedAttributes) {
+                sb.append(SAM_FIELD_SEPARATOR);
+                sb.append(key
+                        + SAM_ATTRIBUTE_SEPARATOR
+                        + GenomicsConverter.getTagType(key)
+                        + SAM_ATTRIBUTE_SEPARATOR
+                        + (keys.contains(key) ? getAttributeAsString(key) : getReadValueForSAMKey(key)));
+            }
+        }
+
+        sb.append("\n");
+        return sb.toString();
+    }
+
+    /**
+     * Generate a list of attributes that should be included in this read's SAM String as attributes
+     * but which are not stored in the attribute list returned by read.getInfo().
+     */
+    private List<String> querySAMAttributes() {
+        final ArrayList<String> attributes = new ArrayList<String>();
+        if (getReadGroup() != null) {
+            attributes.add("RG");
```

Comment:
We also need special handling for MQ which is the mate mapping quality.  It's stored in the Read's mate LinearAlignment field but in sam records it's a bam tag.

Likewise mate cigar MC needs special handling since it's a first class member of Read